### PR TITLE
Capture console statements as one unit including stack trace

### PR DESCRIPTION
**Pull request by sokra**
### Description

fixes the deduplication issue with multiple output

### Testing Instructions

[X] Auto label

### CODE REVIEW

File: crates/next-core/js/src/ipc/index.ts [hunk 1]
```diff
@@ -142,3 +142,39 @@ export const IPC = createIpc<unknown, unknown>(parseInt(PORT, 10));
...
+  // @ts-ignore
+  const original = console[name];
+  // @ts-ignore
+  const stdio = process[stream];
+  // @ts-ignore
+  console[name] = (...args: any[]) => {
+    stdio.write(`TURBOPACK_OUTPUT_B\n`);
+    original(...args);
+    if (addStack) {
+      const stack = new Error().stack?.replace(/^.+\n.+\n/, "") + "\n";
```
Comment (jridgewell) [inline crates/next-core/js/src/ipc/index.ts:156]: Should we use a structured stack here?

File: crates/turbopack-node/js/src/ipc/index.ts [hunk 2]
```diff
@@ -142,3 +142,39 @@ export const IPC = createIpc<unknown, unknown>(parseInt(PORT, 10));
...
+      stdio.write(stack);
+    }
+    stdio.write("TURBOPACK_OUTPUT_E\n");
+  };
+};
+
+improveConsole("error", "stderr", true);
+improveConsole("warn", "stderr", true);
+improveConsole("count", "stdout", true);
+improveConsole("count", "stdout", true);
```
Comment (jridgewell) [inline crates/turbopack-node/js/src/ipc/index.ts:167]: Ditto.

File: crates/turbopack-node/src/pool.rs [hunk 3]
```diff
@@ -167,7 +234,16 @@ impl NodeJsPoolProcess {
                         .unwrap()
                         .read_to_end(&mut stderr)
                         .await?;
-                    Ok((String::from_utf8(stdout)?, String::from_utf8(stderr)?))
+                    fn clean(buffer: Vec<u8>) -> Result<String> {
+                        Ok(String::from_utf8(buffer)?
+                            .lines()
+                            .filter(|line| {
+                                line.len() != MARKER_STR.len() + 1 && !line.starts_with(MARKER_STR)
```
Comment (jridgewell) [inline crates/turbopack-node/src/pool.rs:241]: Why the `line.len() != MARKER_STR.len() + 1 ` condition?

**Review by jridgewell — APPROVED** [review]

File: crates/turbopack-node/src/pool.rs (hunk 3 above)
Comment (sokra) [reply crates/turbopack-node/src/pool.rs:241]: Fast check for the right len

File: crates/next-core/js/src/ipc/index.ts (hunk 1 above)
Comment (sokra) [reply crates/next-core/js/src/ipc/index.ts:156]: Yep, we should do that in the next iteration when you actually do something valueable with the stack trace.

**Pull request merged**